oxen-io/session-android
# [SES-1486] Short voice message fix

### Contributor checklist

 * Virtual Pixel 3a, Android 9 API 28
 * Virtual Pixel 3a, Android 14 API 34

----------

### Description
Users could sent very short (sub 1-second) voice messages - which is of little use & likely undesirable. This PR prevents sending such short voice messages and displays a toast using a string from the Strings project to inform the user that they should press and hold the microphone button to record voice messages.

Due to the fact that the user can manipulate the UI faster than we can set-up and tear-down the audio recording mechanism there is some state juggling going on to keep track of what's happening internally. From my own testing, the mechanism in this PR is robust against button mashing and multi-finger weirdness.

This PR fixes the following issue: https://github.com/oxen-io/session-android/issues/1522

To test you might try to:
- Press and release the microphone button to record an illegally brief voice message (anything < 1 second will show a toast instead), and
- Press & drag up to lock in one fluid motion to quickly lock to recording a voice message (no toast shown on subsequent cancel or send),
- Cancelling out of recording a voice message should work as before (no message sent).

##Note
Following discussion w/ Rebecca / Kee the code has been adjusted to merely log any network connectivity issues for the time being rather than inform the user. This will be adjusted again when SES-2319 has been performed.
